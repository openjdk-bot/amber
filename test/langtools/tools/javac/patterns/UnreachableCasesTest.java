/*
 * @test /nodynamiccopyright/
 * @summary Ensure that unreachable cases of a pattern switch are diagnosed.
 * @compile/fail/ref=UnreachableCasesTest.out -XDrawDiagnostics UnreachableCasesTest.java
 */

public class UnreachableCasesTest {

    class X {}
    class Y extends X {}
    class Z extends Y {}

    public static void main(String [] args) {

        switch (10) {
            case Integer i:
                System.out.println("i = " + i);
                break;
            default:
                System.out.println("Unreachable");
                break;
        }

        Y y = null;
        switch (y) {
            case Z z1: break;
            case Y y1: break;
            case X x1: break;
        }
    }
}