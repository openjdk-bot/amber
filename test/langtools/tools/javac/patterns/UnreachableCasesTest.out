UnreachableCasesTest.java:28:13: compiler.err.pattern.dominated
1 error